pkg/util/mount/mount_linux.go

```diff
@@ -1125,24 +1169,44 @@ func doSafeOpen(pathname string, base string) (int, error) {
 	}()
 
 	currentPath := base
+	lastSegmentIndex := len(segments) - 1
 
 	// Follow the segments one by one using openat() to make
 	// sure the user cannot change already existing directories into symlinks.
-	for _, seg := range segments {
+	for i, seg := range segments {
 		currentPath = filepath.Join(currentPath, seg)
 		if !pathWithinBase(currentPath, base) {
-			return -1, fmt.Errorf("path %s is outside of allowed base %s", currentPath, base)
+			return -1, defaultDeviceInfo, fmt.Errorf("path %s is outside of allowed base %s", currentPath, base)
+		}
+		var deviceStat unix.Stat_t
+		deviceStatErr := unix.Fstatat(parentFD, seg, &deviceStat, unix.AT_SYMLINK_NOFOLLOW)
```

Comment:
I think SafeMakeDir also needs checks for these.  findExistingPrefix also does Openat in a loop

Reply:
So I think if we are going to use `Fstatat` in `findExistingPrefix` then there is no need to use `Openat` there because all we are interesting in knowing is - first directory that does not exist. We don't need `Openat` for that.